source/Nuke.Common/Tools/Unity/UnityTasks.cs

```diff
@@ -35,36 +35,36 @@ private static string GetToolPathViaManualInstallation()
         {
             return EnvironmentInfo.Platform switch
             {
-                PlatformFamily.Windows => $@"{EnvironmentInfo.SpecialFolder(
-                    EnvironmentInfo.Is32Bit
-                        ? SpecialFolders.ProgramFilesX86
-                        : SpecialFolders.ProgramFiles)}\Unity\Editor\Unity.exe",
+                PlatformFamily.Windows => $@"{GetProgramFiles()}\Unity\Editor\Unity.exe",
                 PlatformFamily.OSX => "/Applications/Unity/Unity.app/Contents/MacOS/Unity",
-                PlatformFamily.Linux => null,
-                PlatformFamily.Unknown => null,
-                _ => null
+                _ => throw new Exception($"Cannot determine Unity installation path for current platform.")
```

Comment:
This method should really return `null`. The generic exception will be thrown [here](https://github.com/matkoch/common/blob/develop/source/Nuke.Common/Tooling/ProcessTasks.cs#L52).

Reply:
Is there a way to throw an exception with a more meaningful message? "ToolPath was not set." doesn't seem too helpful, since it doesn't provide any details (which is the platform being unsupported in this particular case).

https://blog.ploeh.dk/2014/12/23/exception-messages-are-for-programmers/

Reply:
I expect developers being able to work with a stacktrace.

Reply:
Ok, you've got a point. "ToolPath was not set." will cause them to set it manually anyway, so it's not that big of a deal.

Should I change `GetToolPathViaHubVersion` to return `null` as well?

Reply:
I think no. In this case the user was using `SetHubVersion`  .. and if that doesn't resolve, it's clearly worth a custom exception.

Reply:
Good point. Pushed a fix.